src/Common/src/Interop/Windows/SChannel/Interop.SecPkgContext_ApplicationProtocol.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal enum ApplicationProtocolNegotiationStatus
+    {
+        None = 0,
+        Success,
+        SelectedClientOnly
+    }
+
+    internal enum ApplicationProtocolNegotiationExt
+    {
+        None = 0,
+        NPN,
+        ALPN
+    }
+
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+    internal class SecPkgContext_ApplicationProtocol
+    {
+        private const int MAX_PROTOCOL_ID_SIZE = 0xFF;
+        public ApplicationProtocolNegotiationStatus NegotiationStatus;
+        public ApplicationProtocolNegotiationExt NegotiationExtension;
+        public byte ProtocolIdSize;
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = MAX_PROTOCOL_ID_SIZE)]
+        public byte[] ProtocolId;
+
+        public byte[] GetProtocolId()
+        {
+            byte[] result = new byte[ProtocolIdSize];
+            Array.Copy(ProtocolId, result, ProtocolIdSize);
+
+            return result;
```

Comment:
Why are we allocating and copying to a new array?  Is the issue that ProtocolIdSize may actually be < ProtocolId.Length?  If so, how about using spans?
```C#
[MarshalAs(...)]
private byte[] _protocolId;

public Span<byte> ProtocolId => new Span<byte>(_protocolId, 0, ProtocolIdSize);
```
?

Reply:
Yeah the size can be less than 255, changed to span.